src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -307,9 +318,15 @@ private void MarkEntireTypeInternal (TypeDefinition type, bool includeBaseTypes,
 
 		void Process ()
 		{
-			while (ProcessPrimaryQueue () || ProcessLazyAttributes () || ProcessLateMarkedAttributes ()) {
+			while (ProcessPrimaryQueue () || ProcessMarkedPending () || ProcessLazyAttributes () || ProcessLateMarkedAttributes ()) {
 
 				// deal with [TypeForwardedTo] pseudo-attributes
+
+				// This marks all exported types that resolve to a marked type. Note that it
+				// may mark unused exported types that happen to resolve to a type which was
+				// marked from a different reference. This seems incorrect.
```

Comment:
We should create an issue for this - otherwise it's just another todo in the code which nobody will ever look at.

Reply:
Added a link to https://github.com/mono/linker/issues/1740